octobot_trading/exchanges/connectors/ccxt_exchange.py

```diff
@@ -301,91 +493,474 @@ async def get_all_currencies_price_ticker(self, **kwargs: dict) -> typing.Option
-            open_orders = await self.get_open_orders(symbol=symbol)
+                self.logger.error(f"Failed to fetch closed orders using get_order_default: Not Supported {e}")
+            except Exception as e:
+                self.logger.error(f"Failed to fetch closed orders using get_order_default: {e}")
+        return None
+
+    async def get_order_using_stop_id(self, order_id: str, symbol: str = None, check_completeness: bool = True,
+                                      **kwargs: dict) -> typing.Optional[dict]:
+        if self.client.has.get('fetchOrder'):
+            try:
+                with self.error_describer():
+                    params = kwargs.pop("params", {})
+                    params["stop_order_id"] = order_id
+                    return await self.parse_order(
+                        await self.client.fetch_order(order_id, symbol, params=params, **kwargs),
+                        check_completeness=check_completeness, )
+            except ccxt.OrderNotFound:
+                # some exchanges are throwing this error when an order is cancelled
+                pass
+            except Exception as e:
+                self.logger.error(f"Failed to get order using get_order_using_stop_id: {e}")
+        return None
+
+    async def get_order_using_get_orders(self, order_id: str, symbol: str = None, check_completeness: bool = True,
+                                         **kwargs: dict
+                                         ) -> typing.Optional[dict]:
+        try:
+            open_orders = await self.get_open_orders(symbol=symbol, check_completeness=check_completeness, )
             for order in open_orders:
                 if order.get(ecoc.ID.value, None) == order_id:
                     return order
+        except Exception as e:
+            self.logger.error(f"Failed to get order using get_order_using_get_orders: {e}")
+        return None
+
+    async def get_order_from_open_and_closed_orders(self, order_id: str, symbol: str = None,
+                                                    check_completeness: bool = True, **kwargs: dict
+                                                    ) -> typing.Optional[dict]:
+        for order in await self.get_open_orders(symbol, check_completeness=check_completeness, **kwargs):
+            if order[ecoc.ID.value] == order_id:
+                return order
+        for order in await self.get_closed_orders(symbol, check_completeness=check_completeness, **kwargs):
+            if order[ecoc.ID.value] == order_id:
+                return order
+        return None  # OrderNotFound
+
+    async def get_order_from_trades(self, order_id, symbol, check_completeness=True):
+        trades = await self.get_my_recent_trades(symbol, check_completeness=check_completeness)
+        # usually the right trade is within the last ones
+        for trade in trades[::-1]:
+            if trade[ecoc.ID.value] == order_id:
+                return trade
```

Comment:
here we are returning a trade instead of an order, maybe we should change the method name

Reply:
i changed it to get_trade